src/java/org/apache/nutch/util/NutchJob.java

```diff
@@ -17,14 +17,16 @@
 
 package org.apache.nutch.util;
 
+import java.io.IOException;
+
 import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapreduce.Job;
 
-/** A {@link JobConf} for Nutch jobs. */
-public class NutchJob extends JobConf {
+/** A {@link Job} for Nutch jobs. */
+public class NutchJob extends Job {
 
-  public NutchJob(Configuration conf) {
-    super(conf, NutchJob.class);
+  public NutchJob(Configuration conf) throws IOException {
+    super(conf, "NutchJob");
```

Comment:
yes, @lewismc if we create a job as NutchJob job = new NutchJob(); then it will be named by default NutchJob and it can be overriden as job.setJobName("<name>"); in the code where the job is being created. Should it be in any other way?